profiling/src/profile/mod.rs

```diff
@@ -686,20 +487,22 @@ impl Profile {
         let mut labels: Vec<Label> = Vec::with_capacity(sample.labels.len());
         let mut local_root_span_id_label_offset: Option<usize> = None;
         for label in sample.labels.iter() {
+            anyhow::ensure!(
+                label.str.is_none() || label.num == 0,
+                "Invalid label: {:?}",
```

Comment:
Do we want to explain in the message why the label is invalid?

Reply:
+1 yes please! I left some suggestions for this on PR https://github.com/DataDog/libdatadog/pull/205

Reply:
I think we can combine both of the ensures into a single one:

```rust
            anyhow::ensure!(
                label.str.is_none() || (label.num == 0 && label.num_unit.is_none()),
                "Invalid label: used both str and num fields: {label:?}"
            );
```

If the `label.str` is none, then it doesn't matter what values are held in `label.num` and `label.num_unit`. If the `label.str` is something, on the other hand, then both `label.num` and `label.num_unit` should be zero-representations. Right?

Reply:
I added some tests to double-check this ^_^ They seem equivalent.

Reply:
Seems good to me! The negations tripped me a bit so I had to pause a bit to mentally parse it, but the logic looks very correct!